Cython/TestUtils.py

```diff
@@ -229,42 +236,96 @@ def unpack_source_tree(tree_file, workdir, cython_root):
     return workdir, header
 
 
+def clear_function_and_Dependencies_caches():
+    Utils.clear_function_caches()
+    Dependencies._dep_tree = None  # discard method caches
+
+
 def fresh_cythonize(*args, **kwargs):
     """Clear function caches and run cythonize"""
     clear_function_and_Dependencies_caches()
     Dependencies.cythonize(*args, **kwargs)
 
 
-def clear_function_and_Dependencies_caches():
-    Utils.clear_function_caches()
-    Dependencies._dep_tree = None  # discard method caches
+def touch_file(path):
+    """Set the access and modified times of the file
+    specified by path as the current time"""
+
+    os.utime(path, None)
+
+
+def relative_lines(lines, line, start, end, fallback=None):
+    """Returns the lines specified by `start` and `end` relative to `line`.
+
+    If `fallback` (Sized Iterable) is specified, it will be used for `line`, `start`, `end`
+    applied to `lines` to generate a message for a ValueError
+    if the original `line` is not found in `lines`."""
+
+    if fallback:
+        if not isinstance(fallback, Sized):
+            raise TypeError("'fallback' must be a Sized instance")
+
+        if not isinstance(fallback, Iterable):
+            raise TypeError("'fallback' must be an instance of the Iterable")
+
+        if len(fallback) != 3:
+            raise ValueError("'fallback' must contain three values"
+                             " for 'line', 'start', 'end'")
+
```

Comment:
I don't think this is necessary. It would fail below if it's something else. That seems enough. This isn't user facing code, it's just test code.
```suggestion
```

Reply:
> It would fail below if it's something else. 

I thought it would be better if it crashes immediately if the `fallback` is something else (` fallback` is not always used), but I think it really isn't that necessary

Reply:
[resolved](https://github.com/cython/cython/pull/4241/commits/cd1aec02767c176ad0e298e12a52b47286924f2f)